client-runtime/src/main/java/com/microsoft/rest/v2/util/ByteBufferUtil.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2.util;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Contains helper methods for dealing with Flowables.
+ */
+public class ByteBufferUtil {
+    /**
+     * Read the contents of the ByteBuffer into a byte[].
+     * @param byteBuffer The ByteBuffer to read from.
+     * @return The created byte[].
+     */
+    public static byte[] toByteArray(ByteBuffer byteBuffer) {
+        byte[] byteArray = null;
+        if (byteBuffer != null) {
+            byteArray = new byte[byteBuffer.remaining()];
+            byteBuffer.get(byteArray);
+        }
+        return byteArray;
+    }
```

Comment:
I forgot to mention this, but next time you touch this class you might want to add a private constructor. Easy to forget and not likely to result in misuse, so it's no big deal.